python-test/src/main/java/com/hazelcast/jet/tests/python/PythonTest.java

```diff
@@ -0,0 +1,133 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.tests.python;
+
+import com.hazelcast.collection.IList;
+import com.hazelcast.internal.nio.IOUtil;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.Sources;
+import com.hazelcast.jet.python.PythonServiceConfig;
+import com.hazelcast.jet.python.PythonTransforms;
+import com.hazelcast.jet.tests.common.AbstractSoakTest;
+import com.hazelcast.map.IMap;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static com.hazelcast.jet.impl.util.Util.uncheckCall;
+import static com.hazelcast.query.Predicates.alwaysTrue;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.IntStream.range;
+
+public class PythonTest extends AbstractSoakTest {
+
+    private static final String SOURCE_MAP_NAME = "pythonSourceMap";
+    private static final String SINK_LIST_NAME = "pythonSinkList";
+    private static final String PYTHON_BASE_DIR = "python_base_dir";
+    private static final String PYTHON_HANDLER_MODULE = "echo";
+    private static final String PYTHON_HANDLER_FUNCTION = "handle";
+    private static final String ECHO_HANDLER_FUNCTION =
+            "def handle(input_list):\n" +
+                    "    return ['echo-%s' % i for i in input_list]\n";
+
+    private static final int DEFAULT_ITEM_COUNT = 10_000;
+
+    private File baseDir;
+    private IMap<Integer, String> sourceMap;
+    private IList<String> sinkList;
+    private int itemCount;
+
+
+    public static void main(String[] args) throws Exception {
+        new PythonTest().run(args);
+    }
+
+    @Override
+    protected void init() throws Exception {
+        baseDir = createTempDirectory();
+
+        sourceMap = jet.getMap(SOURCE_MAP_NAME);
+        sinkList = jet.getList(SINK_LIST_NAME);
+
+        sourceMap.clear();
+        sinkList.clear();
+
+        itemCount = propertyInt("itemCount", DEFAULT_ITEM_COUNT);
+
+        installFileToBaseDir();
+        range(0, itemCount)
+                .mapToObj(i -> sourceMap.putAsync(i, String.valueOf(i)).toCompletableFuture())
+                .collect(toList())
+                .forEach(f -> uncheckCall(f::get));
+    }
+
+    @Override
+    protected void test() {
+        long begin = System.currentTimeMillis();
+        long jobCount = 0;
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            jet.newJob(pipeline()).join();
+
+            assertEquals(itemCount, sinkList.size());
+            assertEquals(itemCount, sinkList.stream().filter(s -> s.startsWith("echo-")).count());
+            sinkList.clear();
+            if (jobCount % 100 == 0) {
```

Comment:
Do we at least roughly know how often this will log?

Reply:
each job takes 7-8 seconds on my local with 2 embedded instances. should be less for the ansible environment so once in 10 minutes or so

Reply:
:+1: